File: beebly web/src/Entity/Livre.php
<?php

namespace App\Entity;


use DateTime;
use DateTimeInterface;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints\Collection;
use Symfony\Component\Serializer\Annotation\Groups;
/**
 * Livre
 *
 * @ORM\Table(name="livre")
 * @ORM\Entity
 * @ORM\Entity(repositoryClass="App\Repository\LivreRepository")
 */
class Livre
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups({"post:read"})
     */
    private $id;

    /**
     * @var string|null
     *
     * @ORM\Column(name="titre", type="string", length=255, nullable=true, options={"default"="NULL"})
     * @Groups({"post:read"})
     */
    private $titre ;

    /**
     * @var string|null
     *
     * @ORM\Column(name="categorie", type="string", length=255, nullable=true, options={"default"="NULL"})
     * @Groups({"post:read"})
     */
    private $categorie ;

    /**
     * @var DateTime|null
     *
     * @ORM\Column(name="date_publication", type="datetime", nullable=true)
     * @Groups({"post:read"})
     */
    private $datepublication;

    /**
     * @var string|null
     *
     * @ORM\Column(name="image", type="string", length=65535, nullable=true, options={"default"="NULL"})
     * @Groups({"post:read"})
     */
    private $image;

    /**
     * @var float
     *
     * @ORM\Column(name="prix", type="float", precision=10, scale=0, nullable=false)
     * @Groups({"post:read"})
     */
    private $prix;

    /**
     * @var string
     *
     * @ORM\Column(name="description_etat_livre", type="string", length=255, nullable=false)
     * @Groups({"post:read"})
     */
    private $descriptionetatlivre;

    /**
     * @var int
     *
     * @ORM\Column(name="note", type="integer", nullable=false)
     * @Groups({"post:read"})
     */
    private $note;
    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="user_id", referencedColumnName="id")
     * })
     * @Groups({"post:read"})
     */
    private $user;

    public function __construct()
    {
        $this->items = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(?string $titre): self
    {
        $this->titre = $titre;

        return $this;
    }

    public function getCategorie(): ?string
    {
        return $this->categorie;
    }

    public function setCategorie(?string $categorie): self
    {
        $this->categorie = $categorie;

        return $this;
    }

    public function getDatePublication(): DateTimeInterface
    {
        return $this->datepublication;
    }

    public function setDatePublication(DateTimeInterface $DatePublication): self
    {
        $this->datepublication = $DatePublication;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(?float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getDescriptionEtatLivre(): ?string
    {
        return $this->descriptionetatlivre;
    }

    public function setDescriptionEtatLivre(?string $DescriptionEtatLivre): self
    {
        $this->descriptionetatlivre = $DescriptionEtatLivre;

        return $this;
    }

    public function getNote(): ?int
    {
        return $this->note;
    }

    public function setNote(?int $note): self
    {
        $this->note = $note;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }
    
    public function setUser(?User $user): self
    {
        $this->user = $user;
    
        return $this;
    }
    

    /**
     * @return Collection<int, Item>
     */
    public function getItems(): Collection
    {
        return $this->items;
    }

    public function addItem(Item $item): self
    {
        if (!$this->items->contains($item)) {
            $this->items->add($item);
            $item->setLivre($this);
        }

        return $this;
    }

    public function removeItem(Item $item): self
    {
        if ($this->items->removeElement($item)) {
            // set the owning side to null (unless already changed)
            if ($item->getLivre() === $this) {
                $item->setLivre(null);
            }
        }

        return $this;
    }

}

File: Mobile Beebly/src/com/mycompany/myapp/entities/Livre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

import java.util.Date;

/**
 *
 * @author sahar
 */
public class Livre {
    
    private int id;
    private String titre;
    private String categorie;
    private Date datePublication;
    private String image;
    private double prix;
    private String descriptionEtatLivre;
    private int note;
    private User idUser;

    public Livre() {
    }

    public Livre(String titre, String categorie, Date datePublication, String image, double prix, String descriptionEtatLivre, int note) {
        this.titre = titre;
        this.categorie = categorie;
        this.datePublication = datePublication;
        this.image = image;
        this.prix = prix;
        this.descriptionEtatLivre = descriptionEtatLivre;
        this.note = note;
    }

    public Livre(int id, String titre, String categorie, Date datePublication, String image, double prix, String descriptionEtatLivre, int note) {
        this.id = id;
        this.titre = titre;
        this.categorie = categorie;
        this.datePublication = datePublication;
        this.image = image;
        this.prix = prix;
        this.descriptionEtatLivre = descriptionEtatLivre;
        this.note = note;
    }

    public User getIdUser() {
        return idUser;
    }

    public void setIdUser(User idUser) {
        this.idUser = idUser;
    }

    public int getId() {
        return id;
    }

    public String getTitre() {
        return titre;
    }

    public String getCategorie() {
        return categorie;
    }

    public Date getDatePublication() {
        return datePublication;
    }

    public String getImage() {
        return image;
    }

    public double getPrix() {
        return prix;
    }

    public String getDescriptionEtatLivre() {
        return descriptionEtatLivre;
    }

    public int getNote() {
        return note;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public void setCategorie(String categorie) {
        this.categorie = categorie;
    }

    public void setDatePublication(Date datePublication) {
        this.datePublication = datePublication;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public void setDescriptionEtatLivre(String descriptionEtatLivre) {
        this.descriptionEtatLivre = descriptionEtatLivre;
    }

    public void setNote(int note) {
        this.note = note;
    }

    @Override
    public String toString() {
        return "Livre{" + "id=" + id + ", titre=" + titre + ", categorie=" + categorie + ", datePublication=" + datePublication + ", image=" + image + ", prix=" + prix + ", descriptionEtatLivre=" + descriptionEtatLivre + ", note=" + note + '}';
    }
    
}
